python/ql/lib/semmle/python/security/dataflow/CleartextLoggingCustomizations.qll

```diff
@@ -40,6 +40,10 @@ module CleartextLogging {
    * A source of sensitive data, considered as a flow source.
    */
   class SensitiveDataSourceAsSource extends Source, SensitiveDataSource {
+    SensitiveDataSourceAsSource() {
+      not SensitiveDataSource.super.getClassification() = SensitiveDataClassification::id()
```

Comment:
Just to double-check, do we want `not ... = ...` here, and not `... != ...`? If I'm not mistaken, the latter would allow something that's both `id` and some other classification, whereas the former would disallow anything that can be seen as `id` (regardless of any other classifications).

Reply:
Not something I had thought about. I agree with your analysis on what the code would do.

Given the problem about cross-talk in dictionaries, I guess we're better off with disallowing anything that can be seen as an `id` :thinking:

Reply:
Makes sense. :+1: